### PR TITLE
refactor. info in src/refactor.txt

### PR DESCRIPTION
**Pull request by lilyzeiset**
lib.rs:
    ALL macro_rules! definitions go here
    run_app()

coordinate.rs:
    anything to do with the coordinate/grid structure
    Coordinate
    Row
    Col

grammar.rs:
    anything to do with grammars
    Grammar
    Kind
    Interactive

model.rs:
    anything to do with the state of the application
     still almost 800 lines, maybe could be split up more
    Model
    SideMenu
    Action
    view_ functions

session.rs:
    all saving/loading code will go here
    Session

style.rs:
    Style

utils.rs:
    all misc/helper function that AREN'T macros go here

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -474,16 +65,14 @@ macro_rules! row_col_vec {
...
 
-#[derive(Parser)]
-#[grammar = "coordinate.pest"]
-struct CoordinateParser;
-
 // macro for easily defining a coordinate
 // either absolutely or relative to it's parent coordinate
 // TODO: this code is messy, can be optimized more later 
+#[macro_export]
 macro_rules! coord {
```
Comment (Korede-TA) [inline src/lib.rs]: can we move the macros related to `Coordinate` into coordinate.rs?

File: src/model.rs [hunk 2]
```diff
@@ -0,0 +1,768 @@
...
+
+    fn to_session(&self) -> Session {
+        Session {
+            root: self.root.clone(),
+            meta: self.meta.clone(),
+            grammars: self.grammars.clone(),
+        }
+    }
+
+    fn get_style(&self, coord: &Coordinate) -> String {
```
Comment (Korede-TA) [inline src/model.rs]: remove this from the model impl and add it to style.rs as a stand-alone function. don’t forget to replace references to “self” with the concrete “Model” type

File: src/model.rs [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+                        });
+                        self.grammars = grammars;
+                    }
+                }
+                true
+            }
+        }
+    }
+
+    fn view(&self) -> Html {
```
Comment (Korede-TA) [inline src/model.rs]: maybe we should also move these `view_` functions out into a new view.rs file ? would cut down the bloat and help us keep all `html!`-related code in one place

File: src/lib.rs (hunk 1 above)
Comment (lilyzeiset) [reply src/lib.rs]: possibly... I had them in utils.rs originally and couldn't get it to work but it should technically be possible.

File: src/model.rs (hunk 2 above)
Comment (lilyzeiset) [reply src/model.rs]: will do.

File: src/model.rs (hunk 3 above)
Comment (lilyzeiset) [reply src/model.rs]: good idea, will do.

**Pull request merged**